S2VX.Game/Editor/SelectToolState.cs

```diff
@@ -1,5 +1,145 @@
+                mousePosX = Math.Clamp(mousePosX, 0, Editor.NotesTimeline.TickBarNoteSelections.DrawWidth); // temp until NoteTimeline Scroll on drag is implemented
+                var relativeMousePosX = mousePosX / Editor.NotesTimeline.TickBarNoteSelections.DrawWidth;
+                var gameTimeDeltaFromMiddle = (relativeMousePosX - 0.5f) * Editor.NotesTimeline.SectionLength * 1000;
+                var gameTimeAtMouse = Story.GameTime + gameTimeDeltaFromMiddle;
+
+                var selectedNoteToTimeCopy = new Dictionary<Note, double>(SelectedNoteToTime);
+                foreach (var noteAndTime in SelectedNoteToTime) {
+                    var note = noteAndTime.Key;
+                    var newTime = GetClosestTickTime(gameTimeAtMouse) + NoteToDragPointDelta[note];
+                    note.EndTime = newTime;
+                    selectedNoteToTimeCopy[note] = newTime;
+                }
+                SelectedNoteToTime = selectedNoteToTimeCopy;
+            }
+        }
+
+        public override void OnToolDragEnd(DragEndEvent _) => DragTimelineNote = false;
+
+        public override void HandleExit() {
+            Editor.NotesTimeline.TickBarNoteSelections.Clear();
+            Editor.NoteSelectionIndicators.Clear();
+        }
+
+        protected override void Update() {
+            Editor.NotesTimeline.TickBarNoteSelections.Clear();
+            var lowerBound = Story.GameTime - Editor.NotesTimeline.SectionLength * 1000 / 2;
+            var upperBound = Story.GameTime + Editor.NotesTimeline.SectionLength * 1000 / 2;
+            foreach (var noteAndTime in SelectedNoteToTime) {
+                if (lowerBound <= noteAndTime.Value && noteAndTime.Value <= upperBound) {
+                    var relativePosition = (noteAndTime.Value - lowerBound) / (Editor.NotesTimeline.SectionLength * 1000);
+                    var indication = new RelativeBox {
+                        Colour = Color4.LimeGreen.Opacity(0.727f),
+                        Width = NotesTimeline.TimelineNoteWidth + 0.009727f,
+                        Height = NotesTimeline.TimelineNoteHeight + 0.1f,
+                        X = (float)relativePosition,
+                        Y = 0.2f,
+                        Anchor = Anchor.TopLeft,
+                        RelativePositionAxes = Axes.Both,
+                        RelativeSizeAxes = Axes.Both,
+                    };
+                    Editor.NotesTimeline.TickBarNoteSelections.Add(indication);
+                }
+            }
+            foreach (var noteSelection in Editor.NoteSelectionIndicators) {
+                noteSelection.Rotation = NoteSelectionToNote[noteSelection].SquareNote.Rotation;
+                noteSelection.X = NoteSelectionToNote[noteSelection].SquareNote.Position.X;
+                noteSelection.Y = NoteSelectionToNote[noteSelection].SquareNote.Position.Y;
+            }
+        }
         public override string DisplayName() => "Select";
     }
-}
+}
```

Comment:
Add a newline at end of file

Reply:
You can enforce this as a code styling rule I think by changing this line to true in the editorconfig: https://github.com/maxrchung/S2VX/blob/bf13eeaff45e940aeb19af020a382698bd4268e9/.editorconfig#L18

Reply:
Ok